dashboard/app/controllers/programming_environments_controller.rb

```diff
@@ -4,7 +4,7 @@ class ProgrammingEnvironmentsController < ApplicationController
 
   before_action :require_levelbuilder_mode_or_test_env, except: [:index, :show, :docs_show, :docs_index]
   before_action :set_programming_environment, except: [:index, :docs_index, :new, :create, :docs_show]
-  authorize_resource
+  authorize_resource except: [:get_summary_by_name]
```

Comment:
The only time a programming environment isn't accessible is if `published` is false, which is the default so that curriculum writers can add new documentation without worrying about it being half-finished. It would be nice to keep that functionality consistent but removing this line and adding `summary_by_name` to [this line](https://github.com/code-dot-org/code-dot-org/blob/5b3803ad4bcfab6a70823e399845e0e3a7d5e8d4/dashboard/app/models/ability.rb#L81).

Reply:
done! I also needed to add [this line](https://github.com/code-dot-org/code-dot-org/pull/45889/files#diff-542e614461454997fa0e42af3564f1b0ed885f6687cbea950d7fc30069f813aaR100). Not sure if we'll need that too for the `docs_...` methods, without it the requests were always authorized.